### DISCUSSION TITLE
Adjust ZADDEDDATE and delete ZIMPORTSESSION

### DISCUSSION THREAD

**Discussion by vandreykiv** [RhetTbull/osxphotos#941] [General]
Hello, 

I'm trying to clean up my library after migration from Google photos. Using osxphotos I was able to match photos with albums, sort them and everything seems fine. Photos were synced to iCloud and now are the same across all devices. 

Next, I want to remove them from import and adjust `added date` to ensure the correct order in recent. Modifying database locally   I was able to archive that but changes not synced across devices. Photos app shows that everything is in sync. 

Is there any easy way to archive this? 

Thanks,
Slava

**Comment by RhetTbull** [comment]
Try setting `ZASSET.ZCLOUDLOCALSTATE = 0` for the assets in question.

I haven't tested this but was thinking about this recently as I think I'll need to do the same for the `osxphotos timewarp` command.

**Comment by vandreykiv** [comment]
Thanks, 

```
SELECT ZCLOUDDELETESTATE FROM ZASSET where ZCLOUDDELETESTATE <> 0
````
shows that all are 0 already.

**Reply by vandreykiv** [reply]
Interesting. On another laptop where I use "optimised storage" option it shows that almost photos has ZCLOUDDELETESTATE=1.

Laptop with all zeros configured to store originals of assets.

**Reply by RhetTbull** [reply]
It may be that ZADDEDDATE isn't synced after initial sync as it might be considered immutable.

**Reply by RhetTbull** [reply]
Did some more digging. It's not `ZCLOUDLOCALSTATE` which indicates where or not item is in the cloud.  I did a bunch of testing with `osxphotos snap`, make a change, `osxphotos diff`.  Changing a value in the local database manually (e.g. via sqlite) does not force a sync to the cloud.  Any change made in Photos creates entries for `ACHANGE` and `ATRANSACTION` tables. I have not done any reverse engineering on these but I believe these encode the changes that need to be sent to the cloud. The data is definitely encoded in some sort of binary format and these are likely difficult to reproduce. However, I did find that making any other change to the photo via Photos itself, did force a sync to the cloud and the manual changes also propagated.  I did not however test ZADDEDDATE directly as I didn't want to mess up my library but I did test ZTIMEZONE and was able to push those changes.

What I did was use [PhotoScript](https://github.com/RhetTbull/PhotoScript) to try changing both the date and the favorite status of the photo and both worked to force a sync to the cloud of the other changes.  

So you could do this:

```python
import PhotoScript
import datetime

uuid = #uuid of photo
photo = photoscript.Photo(uuid)
photo.date = photo.date + datetime.timedelta(seconds=1)
photo.date = photo.date + datetime.timedelta(seconds=-1)

# or
photo.favorite = not photo.favorite
photo.favorite = not photo.favorite
```

Then see if the changes sync.

**Reply by vandreykiv** [reply]
I tried this way but seems that ZADDEDDATE doesn't sync in this case. Seems like I have to live with messed recents album :( 

Thanks for your help
